crates/rpc/src/server/api.rs

```diff
@@ -10,12 +10,10 @@ use miden_node_proto::generated::{self as proto};
 use miden_node_proto::try_convert;
 use miden_node_utils::ErrorReport;
 use miden_node_utils::limiter::{
-    QueryParamAccountIdLimit,
-    QueryParamLimiter,
-    QueryParamNoteIdLimit,
-    QueryParamNoteTagLimit,
+    QueryParamAccountIdLimit, QueryParamLimiter, QueryParamNoteIdLimit, QueryParamNoteTagLimit,
     QueryParamNullifierLimit,
 };
+use miden_node_utils::panic::handle_panic;
```

Comment:
@Mirko-von-Leipzig initial thinking is that all external-facing RPC endpoints should handle panics. Unsure if we want interal ones to. WDYT?

Reply:
Thinking on it some more; I think we already handle panics in most cases since we spawn tasks and then join them and then exit or restart them.

Exceptions to this would be any axum/tonic servers and anywhere we might spawn and detach a background task. I'm not aware of any of the latter, and that wouldn't be good practice.

So maybe its just
- add middleware for the servers
  - it might be good enough to hook into the error tracing middleware to log things - depends how they propagate errors/panics :shrug: 
- register a global panic handler (and for tokio?), but check that it only captures unhandled panics

Reply:
Yea for example from ntx builder:
```
Error: Component ntx-builder failed

Caused by:
    0: Joining component task
    1: task 20 panicked with message "attempt to subtract with overflow"
```

Reply:
So panics captured by tokio should be handled by us as errors already. Is there any part of the stack where you think we may have not handled such cases appropriately?

Reply:
The only _known_ unlogged cases would be the http/grpc servers.

All other tasks/threads _should_ be joined and handled/aborted somewhere. The only reason to consider an additional global panic handler (and tokio panic handler) is as a backup for unknowns.

Reply:
Have updated to use set_hook. I don't follow what you mean by tokio panic handler or why that would be required (tokio should catch panics for us and return as errors which we handle already).

Reply:
>  I don't follow what you mean by tokio panic handler or why that would be required (tokio should catch panics for us and return as errors which we handle already).

Because that's only true if we keep and join the handle. A common pattern is to spawn and forget the task at which point it will just do... what? I hope it still triggers the panic hook?